File: server/datasets/python_source-code/Multiple-A3/organic_7.py
# Linked List Reversal - organic submission 7
# Plain Node objects + a standalone function, no wrapper class at all, with
# a basic assert-based self-test.

class Node:
    def __init__(self, value):
        self.value = value
        self.next = None


def make_chain(values):
    head = tail = None
    for v in values:
        node = Node(v)
        if head is None:
            head = node
        else:
            tail.next = node
        tail = node
    return head


def reverse_chain(head):
    prev = None
    node = head
    while node:
        tmp = node.next
        node.next = prev
        prev = node
        node = tmp
    return prev


def chain_to_list(head):
    out = []
    while head:
        out.append(head.value)
        head = head.next
    return out


def _self_test():
    head = make_chain([1, 2, 3])
    reversed_head = reverse_chain(head)
    assert chain_to_list(reversed_head) == [3, 2, 1]
    print("self-test passed")


if __name__ == "__main__":
    _self_test()
    head = make_chain([9, 7, 5, 3, 1])
    print("before:", chain_to_list(head))
    head = reverse_chain(head)
    print("after: ", chain_to_list(head))

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_a3/organic_7.java
// Linked List Reversal - organic submission 7
// Plain Node objects + standalone functions + self-test.

class Node {
    int value;
    Node next;

    public Node(int value) {
        this.value = value;
        this.next = null;
    }
}

class Solution {
    public static Node makeChain(int[] values) {
        Node head = null;
        Node tail = null;
        for (int v : values) {
            Node node = new Node(v);
            if (head == null) {
                head = node;
            } else {
                tail.next = node;
            }
            tail = node;
        }
        return head;
    }

    public static Node reverseChain(Node head) {
        Node prev = null;
        Node node = head;
        while (node != null) {
            Node tmp = node.next;
            node.next = prev;
            prev = node;
            node = tmp;
        }
        return prev;
    }

    public static void printChain(Node head) {
        Node cur = head;
        StringBuilder sb = new StringBuilder();
        while (cur != null) {
            sb.append(cur.value);
            if (cur.next != null) sb.append(" ");
            cur = cur.next;
        }
        System.out.println(sb.toString());
    }

    public static void main(String[] args) {
        Node head = makeChain(new int[]{9, 7, 5, 3, 1});
        System.out.print("before: ");
        printChain(head);
        head = reverseChain(head);
        System.out.print("after:  ");
        printChain(head);
    }
}
